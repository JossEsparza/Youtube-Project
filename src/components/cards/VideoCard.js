import React from 'react';
import './VideoCard.scss';

const VideoCard = ({ title, description, thumbnails }) => {
  const { url } = thumbnails.high;

  return (
    <div className="card_container">
      <img src={url} alt={title} />
      <h3>{title}</h3>
      <span>{description}</span>
    </div>
  );
};

export default VideoCard;
