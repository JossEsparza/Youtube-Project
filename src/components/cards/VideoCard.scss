.card_container {
  display: flex;
  flex-direction: column;
  margin: 20px;
  width: 300px;
}

.dark {
  h3 {
    color: #cbcccd;
  }
  span {
    color: #a3a3a6;
  }
}

.light {
  h3 {
    color: #383f44;
  }
  span {
    color: #383f44;
  }
}
