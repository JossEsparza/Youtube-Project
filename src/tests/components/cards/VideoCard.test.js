import { shallow } from 'enzyme';
import VideoCard from '../../../components/cards/VideoCard';

describe('Tests the Video Card', () => {
  test('should be displayed correctly', () => {
    const id = {
      videoId: 'ToI6UOZZSzo',
    };

    const snippet_info = {
      title: 'Djokovic Vs Monfils; Alcaraz, Murray & Shapovalov In Action',
      description: 'Highlights of a blockbuster day in Madrid',
      thumbnails: {
        high: {
          url: 'https://i.ytimg.com/vi/ToI6UOZZSzo/mqdefault.jpg',
        },
      },
    };
    const wrapper = shallow(<VideoCard id={id} snippet={snippet_info} />);
    expect(wrapper).toMatchSnapshot();
  });
});
