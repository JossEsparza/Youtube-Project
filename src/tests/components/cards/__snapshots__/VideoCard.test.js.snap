// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Video Card should be displayed correctly 1`] = `ShallowWrapper {}`;
